### PR TITLE
Fix Machine ID GitHub Discover flow

### PR DESCRIPTION
**Pull request by strideynet**
Fixed three problems with this:
1. Made the linux user a required field - without this you can't actually SSH into anything
2. Removed the unneeded Kubernetes fields from the role.
3. Fixed the indentation in the YAML for the node_labels. A single set of label would be dropped due to wrong identing, and more than one would cause the flow to crash.

changelog: Fixed the Machine ID and GitHub Actions wizard.

### CODE REVIEW

**Review by ryanclark — APPROVED** [review]

**Review by mcbattirola — APPROVED** [review]
Thanks! 🙏

File: web/packages/teleport/src/Bots/Add/GitHubActions/ConfigureBot.tsx [hunk 1]
```diff
@@ -181,6 +181,7 @@ export function ConfigureBot({ nextStep, prevStep }: FlowStepProps) {
                     login: e.target.value,
                   })
                 }
+                rule={requireValidSSHUser}
```
Comment (gzdunek) [inline web/packages/teleport/src/Bots/Add/GitHubActions/ConfigureBot.tsx:184]: I wanted to suggest using a `requiredField` validation rule from `shared/components/Validation/rules`, but I noticed it doesn't trim the input. Was it the reason to add your own function?

**Review by gzdunek — APPROVED** [review]

File: web/packages/teleport/src/Bots/Add/GitHubActions/ConfigureBot.tsx (hunk 1 above)
Comment (strideynet) [reply web/packages/teleport/src/Bots/Add/GitHubActions/ConfigureBot.tsx:184]: Actually, I just had no clue we had shared helpers for that. I'll switch it in.

Comment (strideynet) [reply web/packages/teleport/src/Bots/Add/GitHubActions/ConfigureBot.tsx]: 4fb0d6f4c6

**Review by ryanclark — APPROVED** [review]

**Pull request merged**